### ISSUE TITLE
docs(whitepaper): tiered storage plugins design

### ISSUE THREAD

**Issue by petermetz** [hyperledger-cacti/cacti#348]
### Description

We need to design storage with tiers instead of focusing on the type of schema the data has (which is also a valid concern, but the tiers are much more important/significant).

Examples of tiers would be:
1. Storage for data relevant/accessible to the API server
2. Storage for data relevant/accessible to the Node (e.g. cluster of API servers)
3. Storage for data relevant/accessible to the Consortium (needs consortium since here the data is shared/accessed among different consortium members who don't necessarily trust each other implicitly.

Context:
There is an initial proposal in the whitepaper that talks about having different packages with plugins for relational data/documents/key value storage so that plugin developers can optimize their code by using the most fitting storage engines. This is cool, but what's much more important is that we give plugins the flexibility to store private data of a consortium member "locally" (in the sense that other members may not need to have access to this data nor it may be relevant to them at all).
In a way a consortium can also be thought of as a large system comprised of micro-services where each member can have their own micro-service with the data held internally to the service and of course some data needed/being available on a global scale for all.

### Acceptance Criteria
1. Whitepaper is updated with a proposed future design of the storage tiers and what they should be storing
2. Example implementations are mentioned as well such as having a global storage plugin powered by a permissionless with multi-sig wallets ledger or a permissioned ledger that is co-operated by the members making up the consortium (albeit separately from the Cactus nodes themselves)

cc: @jonathan-m-hamilton @takeutak @sfuji822

**Comment by RafaelAPB** [comment]
I agree with this. Level 3 is particularly important for allowing Cactus to enable cross-chain use cases across organizations. And for this, we might need decentralized data storage (acceptance criteria 2).
Now, this could be implemented by whoever implements the first version of ODAP in Cactus, as it requires changes to the consortium management and its underlying data storage.

Could be more feasible to move this feature to v1.0.0, if it is delivered along with ODAP.

**Comment by petermetz** [comment]
@RafaelAPB All agreed and all help is welcome so thank you very much in advance for this one as well! :-) Added this to the v1.0.0 milestone.

**Comment by RafaelAPB** [comment]
Happy to discuss this on the meetings!

**Comment by RafaelAPB** [comment]
We are studying the option to create a new whitepaper as part of https://github.com/hyperledger-cacti/cacti/issues/4037

Thus, this issue is closed.
